mantle/platform/qemu.go

```diff
@@ -101,44 +101,37 @@ type Disk struct {
 	nbdServCmd     exec.Cmd // command to serve the disk
 }
 
-// ParseDiskSpec converts a disk specification into a Disk. The format is:
-// <size>[:<opt1>,<opt2>,...].
-func ParseDiskSpec(spec string) (*Disk, error) {
-	split := strings.Split(spec, ":")
-	var size string
+func ParseDisk(spec string) (*Disk, error) {
+	var diskmap map[string]string
+	var disksize int64
 	var channel string
-	multipathed := false
 	sectorSize := 0
 	serial_opt := []string{}
-	if len(split) == 1 {
-		size = split[0]
-	} else if len(split) == 2 {
-		size = split[0]
-		for _, opt := range strings.Split(split[1], ",") {
-			if opt == "mpath" {
-				multipathed = true
-			} else if opt == "4k" {
-				sectorSize = 4096
-			} else if strings.HasPrefix(opt, "channel=") {
-				channel = strings.TrimPrefix(opt, "channel=")
-				if channel == "" {
-					return nil, fmt.Errorf("invalid channel opt %s", opt)
-				}
-			} else if strings.HasPrefix(opt, "serial=") {
-				serial := strings.TrimPrefix(opt, "serial=")
-				if serial == "" {
-					return nil, fmt.Errorf("invalid serial opt %s", opt)
-				}
-				serial_opt = []string{"serial=" + serial}
-			} else {
-				return nil, fmt.Errorf("unknown disk option %s", opt)
-			}
+	multipathed := false
+	var err error
+
+	if disksize, diskmap, err = util.ParseDiskSpec(spec); err != nil {
```

Comment:
Very minor style thing, usually in cases like this I do:

```
disksize, diskmap, err := util.ParseDiskSpec(spec)
if err != nil {
...
}
```

because this avoids the need to "pre-declare" the variables above.